packages/agents-hosting/src/auth/msalConnectionManager.ts

```diff
@@ -90,7 +91,14 @@ export class MsalConnectionManager implements Connections {
    * ServiceUrl is:  A regex to match with, or "*" for any serviceUrl value.
    * Connection is: A name in the 'Connections' list.
    */
-  getTokenProvider (audience: string, serviceUrl: string): MsalTokenProvider {
+  getTokenProvider (identity: JwtPayload, serviceUrl: string): MsalTokenProvider {
+    let audience
+    if (Array.isArray(identity.aud)) {
+      audience = identity.aud[0]
+    } else {
+      audience = identity.aud
+    }
+
     if (!audience || !serviceUrl) throw new Error('Audience and Service URL are required to get the token provider.')
```

Comment:
We should probably change this error message, as the audience is not the parameter anymore.

Reply:
Pushed a change.  See if you agree with that.

Reply:
Looks good.